### PR TITLE
Update switch to navigate

### PR DESCRIPTION
**Pull request by cwarnermm**
#### Summary
Updated instances of "switch" to "navigate".

#### Ticket Link
Working from a Boards ticket.

#### Release Note
```release-note
Updated instances of "switch" to "navigate".
```

### CODE REVIEW

**Review by michelengelen — APPROVED** [review]
LGTM ... you can fix the failing snapshot test by running `npm run updatesnapshot` in your terminal and commit those changes to the branch!

**Comment by cwarnermm** [comment]
Thanks, @michelengelen! That command took 4494 seconds to complete and tied up all available resources on my machine while it ran. Does it typically take that long for you too (or is my machine underpowered?)

**Comment by michelengelen** [comment]
@cwarnermm that is certainly a bit much ... it does take a while, but it should be done in 5-10 minutes at most

**Comment by jgilliam17** [comment]
Thanks @cwarnermm - changes look good on the test server
There are few cypress tests failing because of the update
See PR results here - [Cypress run for PR 9275](https://mattermost-cypress-report.s3.amazonaws.com/53077-47ffb97-ee-ent-webapp-pr-9275-mm-ee-test:47ffb97/mochawesome.html) can be compared here to [master results ](https://mattermost-cypress-report.s3.amazonaws.com/52978-ee-ent-master-mm-ee-test:test/mochawesome.html)

**Comment by cwarnermm** [comment]
Thanks, @jgilliam17! I'm new to our testing frameworks and could some guidance. Would you be open to walking me through necessary changes to our Cypress tests. I'm able to review both the failing tests and the master results, but I'm unclear on how to begin fixing the tests to resolve the failures.

**Comment by jgilliam17** [comment]
Unfortunately, I'm not of much help 🙁 when dealing with Cypress, beyond running and verifying results.
@saturninoabril can you please add the E2E test files that need to be updated to this PR? Thanks

**Comment by lindy65** [comment]
Hi @cwarnermm @jgilliam17 - quick question... if we're changing terminology in the actual channel switcher to Navigate Channels, should we not also change "Find Channel" in the LHS to Navigate Channels?  Just a thought...

image

**Comment by cwarnermm** [comment]
Great question, @lindy65! I'm currently drawing a distinction between the channel switcher component in the Global Header, and the hint text displayed in the LHS. 

For the channel component, we're actively moving away from "switch" terminology in the product UI that could imply a change in the user's context. We've been adopting an "explore" mindset instead (as "navigate"). 

For the hint text itself, I was aiming to:
a) introduce feature parity between the webapp and the mobile app (where mobile had "jump" terminology that's too similar to "switch" for comfort).
b) use the simplest, most common "explore" term for global non-ESL audiences (as "find" rather than "navigate").

Given this context, what do you think? Looking forward to your feedback on this! CC @esethna @justinegeffen @matthewbirtch (all feedback is welcome)

**Comment by matthewbirtch** [comment]
@cwarnermm I'm in favour of moving away from 'Switch', but I think we should be consistent in the modal title to use "Find Channels" there as well. For what it's worth, mobile app v2 will align with the web app and use "Find Channels".

**Comment by lindy65** [comment]
I'm inclinded to agree with Matt on using "Find" across all platforms (instead of Switch on desktop/web and Jump on mobile)

**Comment by jgilliam17** [comment]
Thank @furqanmlk for updating and running E2Es
@cwarnermm will you be making any additional changes per Matt's suggestion or will that be addressed in a separate PR? Let me know - thanks 🙂

**Comment by cwarnermm** [comment]
@jgilliam17 - I'll be applying Matt and Lindy's suggestions for consistency in this PR. I'll notify you once the changes are ready. Thanks!
@furqanmlk - Thank you for updating E2E tests for this update!

File: components/quick_switch_modal/__snapshots__/quick_switch_modal.test.tsx.snap [hunk 1]
```diff
@@ -45,7 +45,7 @@ exports[`components/QuickSwitchModal should match snapshot 1`] = `
     >
       <h1>
         <MemoizedFormattedMessage
-          defaultMessage="Switch Channels"
+          defaultMessage="Navigate Channels"
```
Comment (cwarnermm) [inline components/quick_switch_modal/__snapshots__/quick_switch_modal.test.tsx.snap:48]:
```suggestion
          defaultMessage="Find Channels"
```

File: components/quick_switch_modal/quick_switch_modal.tsx [hunk 2]
```diff
@@ -151,7 +151,7 @@ export default class QuickSwitchModal extends React.PureComponent<Props, State>
             <h1>
                 <FormattedMessage
                     id='quick_switch_modal.switchChannels'
-                    defaultMessage='Switch Channels'
+                    defaultMessage='Navigate Channels'
```
Comment (cwarnermm) [inline components/quick_switch_modal/quick_switch_modal.tsx:154]:
```suggestion
                    defaultMessage='Find Channels'
```

File: e2e/cypress/integration/accessibility/accessibility_sidebar_spec.js [hunk 3]
```diff
@@ -76,7 +76,7 @@ describe('Verify Accessibility Support in Channel Sidebar Navigation', () => {
         cy.focused().tab();
 
         // * Verify if the Plus button has focus
-        cy.findByRole('button', {name: 'Channel Switcher'}).should('be.focused');
+        cy.findByRole('button', {name: 'Navigate Channels'}).should('be.focused');
```
Comment (cwarnermm) [inline e2e/cypress/integration/accessibility/accessibility_sidebar_spec.js:79]:
```suggestion
        cy.findByRole('button', {name: 'Find Channels'}).should('be.focused');
```

File: e2e/cypress/integration/keyboard_shortcuts/esc_close_modal_spec.js [hunk 4]
```diff
@@ -24,7 +24,7 @@ describe('Keyboard Shortcuts', () => {
         const searchTerm = 'test';
 
         // # Open Channel switcher modal by click on the button
-        cy.findByRole('button', {name: 'Channel Switcher'}).click();
+        cy.findByRole('button', {name: 'Navigate Channels'}).click();
```
Comment (cwarnermm) [inline e2e/cypress/integration/keyboard_shortcuts/esc_close_modal_spec.js:27]:
```suggestion
        cy.findByRole('button', {name: 'Find Channels'}).click();
```

File: e2e/cypress/integration/messaging/message_draft_then_switch_channel_spec.js [hunk 5]
```diff
@@ -46,7 +46,7 @@ describe('Message Draft and Switch Channels', () => {
         cy.typeCmdOrCtrl().type('K', {release: true});
 
         // * Verify that the switch model is shown
-        cy.findAllByRole('dialog').first().findByText('Switch Channels').should('be.visible');
+        cy.findAllByRole('dialog').first().findByText('Navigate Channels').should('be.visible');
```
Comment (cwarnermm) [inline e2e/cypress/integration/messaging/message_draft_then_switch_channel_spec.js]:
```suggestion
        cy.findAllByRole('dialog').first().findByText('Find Channels').should('be.visible');
```

File: e2e/cypress/support/ui/sidebar_left.js [hunk 6]
```diff
@@ -96,7 +96,7 @@ Cypress.Commands.add('uiAddDirectMessage', () => {
 });
 
 Cypress.Commands.add('uiGetChannelSwitcher', () => {
-    return cy.get('#lhsNavigator').findByRole('button', {name: 'Channel Switcher'});
+    return cy.get('#lhsNavigator').findByRole('button', {name: 'Navigate Channels'});
```
Comment (cwarnermm) [inline e2e/cypress/support/ui/sidebar_left.js:99]:
```suggestion
    return cy.get('#lhsNavigator').findByRole('button', {name: 'Find Channels'});
```

File: i18n/en.json [hunk 7]
```diff
@@ -3952,7 +3952,7 @@
   "quick_switch_modal.help_mobile": "Type to find a channel.",
   "quick_switch_modal.help_no_team": "Type to find a channel. Use **UP/DOWN** to browse, **ENTER** to select, **ESC** to dismiss.",
   "quick_switch_modal.input": "quick switch input",
-  "quick_switch_modal.switchChannels": "Switch Channels",
+  "quick_switch_modal.switchChannels": "Navigate Channels",
```
Comment (cwarnermm) [inline i18n/en.json:3955]:
```suggestion
  "quick_switch_modal.switchChannels": "Find Channels",
```

File: i18n/en.json [hunk 8]
```diff
@@ -4142,7 +4142,7 @@
   "sidebar_left.add_channel_dropdown.invitePeopleExtraText": "Add people to the team",
   "sidebar_left.channel_filter.filterByUnread": "Filter by unread",
   "sidebar_left.channel_filter.showAllChannels": "Show all channels",
-  "sidebar_left.channel_navigator.channelSwitcherLabel": "Channel Switcher",
+  "sidebar_left.channel_navigator.channelSwitcherLabel": "Navigate Channels",
```
Comment (cwarnermm) [inline i18n/en.json:4145]:
```suggestion
  "sidebar_left.channel_navigator.channelSwitcherLabel": "Find Channels",
```

**Review by michelengelen — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thanks @cwarnermm 
Tested, looks good to merge.
- Verified terminology change Switch Channels to Find Channels and keyboard shortcut text for channel and team navigation.

**Comment by jgilliam17** [comment]
@cwarnermm I didn't merge, please take a look - there are still some checks that need to be addressed.

**Review by michelengelen — APPROVED** [review]

**Comment by cwarnermm** [comment]
@jgilliam17 - I believe this PR is now ready for merge, thanks to @AshishDhama and @michelengelen!

**Comment by cwarnermm** [comment]
Product documentation updated for this product change via https://github.com/mattermost/docs/pull/5213

**Pull request merged**